src/index.js

```diff
@@ -249,145 +243,151 @@ class TerserPlugin {
         []
       );
       const files = [].concat(additionalChunkAssets).concat(chunksFiles);
-      const tasks = [];
 
-      files.forEach((file) => {
-        if (!matchObject(file)) {
-          return;
-        }
+      function* tasks() {
+        const existingAssets = new Set(
```

Comment:
Not sure it is good idea, one each task we will create new Set, in webpack@5 it is already `Set` so better uses `compilation.assets`

Reply:
I don't think this is right. Only one `Set` is generated for a `compilation` (not for every task). In fact, before this change was made, a new `Array` was generated for every task. I explain this more in the commit 35f310c0a22da98bc459a7d8729ff1cb2879c2bc

`compilation.assets` is not a `Set` in webpack@5. You might be thinking of `compilation.modules` or `compilation.chunks`.